integration-cli/docker_cli_run_unix_test.go

```diff
@@ -205,12 +205,36 @@ func (s *DockerSuite) TestRunWithCpuPeriod(c *check.C) {
 func (s *DockerSuite) TestRunWithKernelMemory(c *check.C) {
 	testRequires(c, kernelMemorySupport)
 
-	dockerCmd(c, "run", "--kernel-memory", "50M", "--name", "test", "busybox", "true")
+	dockerCmd(c, "run", "--kernel-memory", "50M", "--name", "test1", "busybox", "true")
 
-	out, err := inspectField("test", "HostConfig.KernelMemory")
+	out, err := inspectField("test1", "HostConfig.KernelMemory")
 	c.Assert(err, check.IsNil)
 	if out != "52428800" {
-		c.Fatalf("setting the kernel memory limit failed")
+		c.Fatalf("setting the kernel memory limit failed, expected 52428800, actual %q", out)
+	}
+
+	dockerCmd(c, "run", "--kernel-memory", "100000K", "--name", "test2", "busybox", "true")
+
+	out, err = inspectField("test2", "HostConfig.KernelMemory")
+	c.Assert(err, check.IsNil)
+	if out != "102400000" {
+		c.Fatalf("setting the kernel memory limit failed, expected 102400000, actual %q", out)
+	}
+
+	dockerCmd(c, "run", "--kernel-memory", "1G", "--name", "test3", "busybox", "true")
```

Comment:
Are there any reasons why this test (1G) and the one above (100000K) are needed?
The first one (50M) is covering positive values and the one below negative with error.
Having the correct value converted between sizes (k,m,g to bytes) is covered with unit tests in pkg/units so I don't think these two tests are needed